src/Simplex/Chat/Types.hs

```diff
@@ -77,6 +89,24 @@ instance ToJSON GroupProfile where toEncoding = J.genericToEncoding J.defaultOpt
 
 instance FromJSON GroupProfile
 
+data GroupInvitation = GroupInvitation
+  { fromMember :: MemberInfo,
+    invitedMember :: MemberInfo,
+    queueInfo :: SMPQueueInfo,
+    groupProfile :: GroupProfile
+  }
+  deriving (Eq, Show)
+
+data ReceivedGroupInvitation = ReceivedGroupInvitation
```

Comment:
why not newtype ReceivedGroupInvitation? these two seem the same

Reply:
newtype can only be used when there one constructor and it has exactly one field - that is the type can have the same internal representation as said field, and therefore can be completely erased and not create an extra wrapper. Here we have many fields.